File: backend/models/alert.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class AlertBase(BaseModel):
    node_id: str
    alert_level: int
    message: str
    channels_sent: str # e.g. "siren,sms,dashboard"

class AlertCreate(AlertBase):
    pass

class Alert(AlertBase):
    id: Optional[str] = Field(None, alias="_id")
    dispatched_at: datetime
    resolved: bool = False
    false_alarm: bool = False
    resolved_at: Optional[datetime] = None

    class Config:
        populate_by_name = True

File: backend/models/Alert.js
const mongoose = require('mongoose');

const AlertSchema = new mongoose.Schema({
  node_id: {
    type: String,
    required: true,
    index: true,
  },
  alert_level: {
    type: Number,
    required: true,
    min: 1,
    max: 5,
  },
  level_name: {
    type: String,
    default: 'Advisory',
  },
  message: {
    type: String,
    required: true,
  },
  dispatched_at: {
    type: Date,
    default: Date.now,
    index: true,
  },
  resolved: {
    type: Boolean,
    default: false,
    index: true,
  },
  resolved_at: {
    type: Date,
    default: null,
  },
  false_alarm: {
    type: Boolean,
    default: false,
  },
  sms_broadcast_count: {
    type: Number,
    default: 0,
  },
  siren_triggered: {
    type: Boolean,
    default: false,
  },
}, {
  timestamps: true,
});

const Alert = mongoose.models.Alert || mongoose.model('Alert', AlertSchema);

module.exports = Alert;
